20150310-wp-core-contrib-day.md

```diff
@@ -0,0 +1,255 @@
+
+## Development
+You're now all set up to develop on WordPress Core, keep your repository up to date with `svn up` and rebuild your WordPress Core site with `grunt` as required or through a `watch` process.
+
+Let's assume you're starting off your WordPress Core contributions by refreshing a patch, this entails downloading the patch file from a ticket, applying it to your checked out source, checking that it still applies correctly and does what is expected against the latest development version of WordPress Core, and then re-creating the patch and attaching to the ticket with an updated name and updated keywords.
+
+### Keep It Clean
+Before you apply a patch, be absolutely sure that your checked out source and build are "clean", there is nothing worse than working on something only to find that you can't create a clean patch because you have some previous code changes from another ticket polluting your code base.
+
+```
+svn diff
+```
+
+If `svn diff` returns anything, maybe because you've already worked on a patch, revert the changes.
+
+```
+svn revert path/to/changed/file
+```
+
+And then double check again with `svn diff`.
+
+When there are no changes in your checkout, make sure you're up to date with other changes that have been committed to the repository.
+
+```
+svn up
+```
+
+Now ensure your cleaned up repository builds, and then [Give It a Spin][].
+
+```
+grunt
+```
+
+### Applying a Patch
+Download the latest patch file from the ticket, it should be named something like: `14584.3.patch`, where `14584` is the ticket number, and `.3` is an optional patch version which is increased on each new patch for a ticket, the first patch file for ticket 14584 would have been named `14584.patch`. If you're refreshing the patch for an older ticket you might find that the patch name is more like `14584v3.diff`.
+
+The file should be copied to the root of your WordPress Core checkout and applied from there with `patch`:
+
+
+```
+patch -p0 < 14584.3.patch
+```
+
+![wordpress-core — Applying a Patch](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/wordpress-core-—-Applying-a-Patch.png)
+
+Patches are always created and applied from the root of the code base, so you should generally see `src/...` or `tests/...` in the file names being patched.
+
+Hopefully the patch applied as expected, if not (maybe that's why you're refreshing it) then you may need to revert anything that partially applied with `svn revert <filename>`.
+
+Because I use the awesome [PhpStorm](https://www.jetbrains.com/phpstorm/) for all my PHP development I can easily find all the changes the patch applied.
+
+![PhpStorm - Changed Files](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/PhpStorm-Changed-Files.png)
```

Comment:
It's not clear what I'm supposed to be looking at in this screenshot. I'm not seeing changes.